assistedInstaller.py

```diff
@@ -48,9 +48,36 @@ def ensure_infraenv_deleted(self, name: str) -> None:
         if name in (x["name"] for x in self.list_infra_envs()):
             self.delete_infra_env(name)
 
-    def download_kubeconfig(self, name: str, path: str, stdout: bool = False) -> None:
-        path = os.path.dirname(path)
-        super().download_kubeconfig(name, path, stdout)
+    def download_kubeconfig_and_secrets(
+        self,
+        name: str,
+        kubeconfig_path: Optional[str] = None,
+        *,
+        log: bool = True,
+    ) -> tuple[str, str]:
+        if kubeconfig_path:
+            kubeconfig_path = os.path.abspath(kubeconfig_path)
+            path = os.path.dirname(kubeconfig_path)
+        else:
+            path = os.path.abspath(os.getcwd())
+
+        real_kubeadminpassword_path = f"{path}/kubeadmin-password.{name}"
+        real_kubeconfig_path = f"{path}/kubeconfig.{name}"
+
+        self.download_kubeconfig(name, path)
+        self.download_kubeadminpassword(name, path)
+
+        if kubeconfig_path and real_kubeconfig_path != kubeconfig_path:
+            # download_kubeconfig() does not support specifying the path full. The caller
+            # requested another name. Rename.
+            os.rename(real_kubeconfig_path, kubeconfig_path)
+            real_kubeconfig_path = kubeconfig_path
+
```

Comment:
Can we just rename the paths every time, has the same result as only renaming if `real_kubeconfig_path != kubeconfig_path`

Maybe we could change the code to
```python
     if kubeconfig_path:
        os.rename(real_kubeconfig_path, kubeconfig_path)
        real_kubeconfig_path = kubeconfig_path

        expected_kubeadminpassword_path = os.path.join(os.path.dirname(kubeconfig_path), f"kubeadmin-password.{os.path.basename(kubeconfig_path)}")
        os.rename(real_kubeadminpassword_path, expected_kubeadminpassword_path)
        real_kubeadminpassword_path = expected_kubeadminpassword_path
```

Reply:
> Can we just rename the paths every time, has the same result as only renaming if real_kubeconfig_path != kubeconfig_path

I don't understand. Do you mean:
```diff
-        if kubeconfig_path and real_kubeconfig_path != kubeconfig_path:
+        if kubeconfig_path:
```

Why is that better?

---

> Maybe we could change the code to

That seems wrong.  For example:

```python
>>> kubeconfig_path='/root/kubeconfig.sno'
>>> os.path.join(os.path.dirname(kubeconfig_path), f"kubeadmin-password.{os.path.basename(kubeconfig_path)}")
'/root/kubeadmin-password.kubeconfig.sno'
```

I don't understand what the purpose of the suggested change is. Could you elaborate?

---

> Only thing is the kubeadmin file is not named like specified, maybe we can rename it as well?

I don't understand "not named like specified". It cannot be specified.

The path of the kubeadmin-password file is not configurable in the patch (beyond the directory name, which is `os.path.dirname(kubeconfig_path)`). Making it configurable seems not useful to me.